### PR TITLE
feat: [OZ] ERC-2612 - Permit

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
Adding tests for [OZ] ERC-2612 - Permit allowing approvals to be made via signatures, as defined in
 * https://eips.ethereum.org/EIPS/eip-2612[ERC-2612].

**Related issue(s)**:
https://github.com/hashgraph/hedera-smart-contracts/issues/564

### CODE REVIEW

File: contracts/solidity/oz/ERC-2612/ERC2612.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (quiet-node) [inline contracts/solidity/oz/ERC-2612/ERC2612.sol:1]: please update license to Apache-2.0

File: contracts/solidity/oz/ERC-2771/ERC2771Forward.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        return (t1, t2, t3, t4);
+    }
+
+    function fund() external payable{}
+
+    function getChainID() external view returns (uint256){
+        return block.chainid;
+    }
+}
```
Comment (quiet-node) [inline contracts/solidity/oz/ERC-2771/ERC2771Forward.sol:31]: extra line EOF

File: test/solidity/oz/ERC-2612/erc-2612.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  return [
+    { name: 'owner', type: 'address' },
+    { name: 'spender', type: 'address' },
+    { name: 'value', type: 'uint256' },
+    { name: 'nonce', type: 'uint256' },
+    { name: 'deadline', type: 'uint256' },
+  ]
+}
+
+describe('@solidityequiv3 ERC2612 Tests', function () {
```
Comment (quiet-node) [inline test/solidity/oz/ERC-2612/erc-2612.js:35]: Please update unit test tag

File: test/solidity/oz/ERC-2612/erc-2612.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  before(async function () {
+    signers = await ethers.getSigners()
+    wallet = signers[0]
+    wallet2 = signers[1]
+
+    const factory = await ethers.getContractFactory('ERC2612Test')
+    contract = await factory.deploy()
+    await contract.deployed()
+    await contract.connect(wallet).mint(MINT_AMOUNT)
+    // const supply = await contract.totalSupply()
```
Comment (quiet-node) [inline test/solidity/oz/ERC-2612/erc-2612.js:52]: Should we remove these comments?

File: test/solidity/oz/erc2771/forward.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        { name: 'to', type: 'address' },
+        { name: 'value', type: 'uint256' },
+        { name: 'gas', type: 'uint256' },
+        { name: 'nonce', type: 'uint256' },
+        { name: 'deadline', type: 'uint48' },
+        { name: 'data', type: 'bytes' },
+    ]
+}
+
+describe('@solidityequiv3 OZ ERC-2771 Fоrward', function () {
```
Comment (quiet-node) [inline test/solidity/oz/erc2771/forward.js:40]: please update unit test tag

File: test/solidity/oz/erc2771/forward.js [hunk 6]
```diff
@@ -0,0 +1,188 @@
+/*-
```
Comment (quiet-node) [inline test/solidity/oz/erc2771/forward.js:1]: Are these files duplicated from erc2771 ticket?

File: test/solidity/oz/erc2771/forward.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+        expect(verifiedStatic[0]).to.be.true
+        expect(verifiedStatic[1]).to.be.false
+        expect(verifiedStatic[2]).to.be.false
+
+        expect(verifiedTrxResult[0]).to.be.true
+        expect(verifiedTrxResult[1]).to.be.false
+        expect(verifiedTrxResult[2]).to.be.false
+    })
+
+})
```
Comment (quiet-node) [inline test/solidity/oz/erc2771/forward.js:188]: extra line EOF

**Review by quiet-node — CHANGES_REQUESTED** [review]
LG but some comments

File: test/solidity/oz/erc2771/forward.js (hunk 6 above)
Comment (stefan-stefanooov) [reply test/solidity/oz/erc2771/forward.js:1]: Kind of. Both tasks are using similar helper functions so I rebased this one from the Forwarder Task.
My plan was to merge the Forwarder first ... so the Comments and Diffs will disappear after we do so.
Thank you for the review :)

**Comment by quiet-node** [comment]
Tests are failing..

**Review by Nana-EC — APPROVED** [review]

**Comment by Nana-EC** [comment]
@stefan-stefanooov please update PR description

**Review by quiet-node — APPROVED** [review]
LG

**Pull request merged**